src/InputToolbar.tsx

```diff
@@ -47,128 +47,96 @@ export interface InputToolbarProps<TMessage extends IMessage> {
-  }
-
-  state = {
-    position: 'absolute',
-  }
-
-  keyboardWillShowListener?: EmitterSubscription = undefined
-  keyboardWillHideListener?: EmitterSubscription = undefined
-
-  componentDidMount() {
-    this.keyboardWillShowListener = Keyboard.addListener(
-      'keyboardWillShow',
-      this.keyboardWillShow,
-    )
-    this.keyboardWillHideListener = Keyboard.addListener(
-      'keyboardWillHide',
-      this.keyboardWillHide,
-    )
-  }
-
-  componentWillUnmount() {
-    if (this.keyboardWillShowListener) {
-      this.keyboardWillShowListener.remove()
+export function InputToolbar<TMessage extends IMessage = IMessage>(props: InputToolbarProps<TMessage>) {
+  const [position, setPosition] = useState('absolute');
+
+  useEffect(() => {
+    const keyboardWillShowListener = Keyboard.addListener('keyboardWillShow', () => setPosition('relative'));
+    return () => keyboardWillShowListener?.remove();
+  }, []);
+
+  useEffect(() => {
+    const keyboardWillHideListener = Keyboard.addListener('keyboardWillHide', () => setPosition('absolute'));
+    return () => keyboardWillHideListener?.remove();
+  }, []);
+
+  const renderActions = useCallbackOne(() => {
+    const { containerStyle, ...rest } = props
+    if (props.renderActions) {
+      return props.renderActions(rest)
+    } else if (props.onPressActionButton) {
+      return <Actions {...rest} />
     }
-    if (this.keyboardWillHideListener) {
-      this.keyboardWillHideListener.remove()
-    }
-  }
-
-  keyboardWillShow = () => {
-    if (this.state.position !== 'relative') {
-      this.setState({
-        position: 'relative',
-      })
-    }
-  }
 
-  keyboardWillHide = () => {
-    if (this.state.position !== 'absolute') {
-      this.setState({
-        position: 'absolute',
-      })
-    }
-  }
-
-  renderActions() {
-    const { containerStyle, ...props } = this.props
-    if (this.props.renderActions) {
-      return this.props.renderActions(props)
-    } else if (this.props.onPressActionButton) {
-      return <Actions {...props} />
-    }
     return null
-  }
+  }, [props.renderActions, props.onPressActionButton]);
 
-  renderSend() {
-    if (this.props.renderSend) {
-      return this.props.renderSend(this.props)
+  const renderSend = useCallbackOne(() => {
+    if (props.renderSend) {
+      return props.renderSend(props)
     }
-    return <Send {...this.props} />
-  }
 
-  renderComposer() {
-    if (this.props.renderComposer) {
-      return this.props.renderComposer(this.props as ComposerProps)
+    return <Send {...props} />
+  }, [props.renderSend]);
+
+  const renderComposer = useCallbackOne(() => {
+    if (props.renderComposer) {
+      return props.renderComposer(props as ComposerProps)
     }
 
-    return <Composer {...this.props} />
-  }
+    return <Composer {...props} />
+  }, [props.renderComposer]);
 
-  renderAccessory() {
-    if (this.props.renderAccessory) {
+  const renderAccessory = useCallbackOne(() => {
```

Comment:
The same message concerning callback returning a component, please move under return.